### PR TITLE
[ATB-217] Reduce cognitive complexity in checkYourPhoneService method

### PR DESCRIPTION
**Pull request by chiarapascucci-gds**
## Proposed changes
Currently the checkYourPhoneService Interface describes an async function that takes in 8 parameters. This is flagged as a code smell as the cognitive complexity is too high (max number of parameters is 7). This PR changes the interface definition so that the method signature only requires 2 parameters

### What changed
The CheckYourPhoneServiceInterface was updated so to take 2 objects as its parameters. Those objects represent the user's input (email, updated info, and otp code) and session details respectively. The way this information is accessed when forming the post request to the relevant API was also changed in line with this restructuring

### Why did it change
Aim is to reduce cognitive complexity of the method

### Issue tracking
- [ATB-217](https://govukverify.atlassian.net/jira/software/c/projects/ATB/boards/241?modal=detail&selectedIssue=ATB-217)

## Testing
Changes were tested during development

### Local
Tested the changes locally (in Docker container) through existing unit and integration tests. All passed
Screenshot 2023-02-06 at 12 22 34

Screenshot 2023-02-06 at 12 20 24

### Deployed
Deployed a new version of the application to the AWS dev environment successfully and manually tested account creation, changing password, account deletion, log in/out. Message received on mobile phone when adding phone number

### Environment variables or secrets
- [x] No environment variables or secrets were added or changed

### Other considerations
- [ ] Demo to a BA, TA, and the team.
- [ ] Recorded demo shared on [#govuk-accounts-tech](https://gds.slack.com/archives/C011Y5SAY3U)
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[ATB-217]: https://govukverify.atlassian.net/browse/ATB-217?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/check-your-phone/check-your-phone-controller.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
 } from "../../utils/validation";
 import { redactPhoneNumber } from "../../utils/strings";
 import xss from "xss";
+import {UpdateInformationInput, UpdateInformationSessionValues} from "../../utils/types";
```
Comment (TaherKapasi) [inline src/components/check-your-phone/check-your-phone-controller.ts:13]:
```suggestion
import { UpdateInformationInput, UpdateInformationSessionValues } from "../../utils/types";
```

File: src/components/check-your-phone/check-your-phone-controller.ts [hunk 2]
```diff
@@ -25,19 +26,26 @@ export function checkYourPhonePost(
...
+      sourceIp: req.ip,
+      sessionId: res.locals.sessionId,
+      persistentSessionId : res.locals.persistentSessionId,
+      userLanguage: xss(req.cookies.lng as string)
+
+    }
+
+    const isPhoneNumberUpdated = await service.updatePhoneNumber(
+        updateInput, sessionDetails
+
```
Comment (TaherKapasi) [inline src/components/check-your-phone/check-your-phone-controller.ts:48]: remove spare line

File: src/components/check-your-phone/check-your-phone-service.ts [hunk 3]
```diff
@@ -1,34 +1,29 @@
 import { getRequestConfig, Http, http } from "../../utils/http";
 import { API_ENDPOINTS, HTTP_STATUS_CODES } from "../../app.constants";
 import { CheckYourPhoneServiceInterface } from "./types";
+import {UpdateInformationInput, UpdateInformationSessionValues} from "../../utils/types";
```
Comment (TaherKapasi) [inline src/components/check-your-phone/check-your-phone-service.ts:4]:
```suggestion
import { UpdateInformationInput, UpdateInformationSessionValues } from "../../utils/types";
```

File: src/components/check-your-phone/tests/check-your-phone-service.test.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { checkYourPhoneService } from "../check-your-phone-service";
 import { expect } from "chai";
 import { API_ENDPOINTS, HTTP_STATUS_CODES } from "../../../app.constants";
 import { getApiBaseUrl } from "../../../config";
+import {UpdateInformationInput, UpdateInformationSessionValues} from "../../../utils/types";
```
Comment (TaherKapasi) [inline src/components/check-your-phone/tests/check-your-phone-service.test.ts:7]:
```suggestion
import { UpdateInformationInput, UpdateInformationSessionValues } from "../../../utils/types";
```

File: src/components/check-your-phone/types.ts [hunk 5]
```diff
@@ -1,12 +1,8 @@
+import {UpdateInformationInput, UpdateInformationSessionValues} from "../../utils/types";
```
Comment (TaherKapasi) [inline src/components/check-your-phone/types.ts:1]:
```suggestion
import { UpdateInformationInput, UpdateInformationSessionValues } from "../../utils/types";
```

**Comment by TaherKapasi** [comment]
Hi @chiarapascucci-gds - could you exclude  the changes from src/assets/javascript/cookies.js please and rebase into single commit please - thanks

**Review by TaherKapasi — CHANGES_REQUESTED** [review]

File: src/components/check-your-phone/check-your-phone-controller.ts [hunk 6]
```diff
@@ -25,19 +26,26 @@ export function checkYourPhonePost(
   service: CheckYourPhoneServiceInterface = checkYourPhoneService()
 ): ExpressRouteFunc {
   return async function (req: Request, res: Response) {
-    const code = req.body["code"];
     const { email, newPhoneNumber } = req.session.user;
-    const { accessToken } = req.session.user.tokens;
 
-    const isPhoneNumberUpdated = await service.updatePhoneNumber(
-      accessToken,
+    const updateInput : UpdateInformationInput= {
```
Comment (KG700) [inline src/components/check-your-phone/check-your-phone-controller.ts:31]:
```suggestion
    const updateInput : UpdateInformationInput = {
```

**Review by chiarapascucci-gds — COMMENTED** [review]
ok

**Review by TaherKapasi — DISMISSED** [review]

File: src/components/check-your-phone/check-your-phone-service.ts [hunk 7]
```diff
@@ -1,34 +1,29 @@
...
-        persistentSessionId,
-        sessionId,
-        userLanguage
+          sessionDetails.accessToken,
+          [HTTP_STATUS_CODES.NO_CONTENT, HTTP_STATUS_CODES.BAD_REQUEST],
+          sessionDetails.sourceIp,
+          sessionDetails.persistentSessionId,
+          sessionDetails.sessionId,
+          sessionDetails.userLanguage
       )
```
Comment (KG700) [inline src/components/check-your-phone/check-your-phone-service.ts:27]: Remove the extra whitespace on lines 10 & 21-26.

**Review by KG700 — APPROVED** [review]

**Review by TaherKapasi — APPROVED** [review]

**Review by huwd — APPROVED** [review]

**Pull request merged**